feat: Ability to use icons for empty workspace.

so-rose opened the pull request:
Hello! I implemented a small feature for own use, which I hope can be of use.

# What is This?
This allows users of ex. `waybar` to use dynamic icons, while still giving empty workspaces an icon instead of a number. This is implemented by renaming new workspaces right when they are created.
This is all behind a config option; defaults remain unchanged.

Implementing this is also an optimization for the rest of the program, which no longer recalculates workspace naming on _every_ `Event::Window`. Instead, only specific `Window` (and now also `Workspace`) events will do so.

**Caveats**:
- There is a small visual artifact when opening a new workspace, when the number gets renamed to the desired icon.. Unfortunately, being a program reacting to sway events, I don't see a way to fix this.

**Future Possibilities**:
- Allowing specific icons for specific numbered (empty) workspaces wouldn't be difficult to add (the idea in #33).
- Which `*.change` types precisely should trigger workspace renaming was determined with common sense, and could do with a one-over. The relevant function is `should_rename_after_event`.

Closes #39.

Let me know if I can change anything that would help it get merged.

pierrechevalier83 commented:
Thank you so much for your contribution. I briefly reviewed the code and it looks good to me in principle. I like that it fixes a usecase that will be valuable for some users while not impacting folks who don't need the feature. I'll want to test the change before merging the PR because an update will immediately be picked up by the AUR, so any regression would break users of AUR package. I don't have time to do this today, but will aim to make some time before next week.

so-rose commented on src/window_manager.rs:
```diff
@@ -114,14 +118,55 @@ pub struct WindowManager {
                 .context("Couldn't connect to WM")?
-                .subscribe(&[EventType::Window])
-                .context("Couldn't subscribe to events of type Window")?,
+                .subscribe(&[EventType::Window, EventType::Workspace])
```
So, this line alone is actually the **only** change that's _strictly_ required to allow for renaming empty workspaces. Every other change in this file isn't strictly needed.

There's a catch: Changing only this makes it so that _every single sway event_ that has anything at all to do with workspaces or windows will trigger a renaming operation of all workspaces.

I felt that was rather wasteful.

so-rose commented:
Thank you so much for the quick reply! I'm glad the code seems reasonable (I'm relatively new to Rust! Let me know if there's anything I can help with.

In the meantime, I fixed the `rustfmt` and `clippy` errors and added some comments to hopefully make the review easier for you.

pierrechevalier83 commented:
Sorry I'm getting back to you so late. I've rebased it since I added hyprland support since then and merged a few other PRs that conflicted. I also did some minor cleanups while I was at it. I'll close this PR for now. Let's discuss what's next on #58
